dd-sdk-android-tv/src/main/kotlin/com/datadog/android/tv/LeanbackViewAttributesProvider.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.tv
+
+import android.view.View
+import androidx.leanback.widget.Action
+import androidx.leanback.widget.ItemBridgeAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.datadog.android.rum.tracking.ViewAttributesProvider
+
+/**
+ * Provides extra attributes for the touch target View.
+ * Those special attributes are specifically related with Android TV [Action].
+ * @see [ACTION_TARGET_ACTION_ID]
+ * @see [ACTION_TARGET_LABEL1]
+ * @see [ACTION_TARGET_LABEL2]
+ */
+class LeanbackViewAttributesProvider :
+    ViewAttributesProvider {
+
+    // region ViewAttributesProvider
+
+    override fun extractAttributes(
+        view: View,
+        attributes: MutableMap<String, Any?>
+    ) {
+        // traverse the target parents
+        var parent = view.parent
+        var child: View? = view
+        while (parent != null) {
+            if (parent is RecyclerView && child != null && isDirectChildOfRecyclerView(child)) {
```

Comment:
Out of curiosity, when can a view, whose parent `is RecyclerView`, not be a direct child of Recycler view? 🤔

Reply:
It happened for me in the past, long time ago when I did the `JetpackViewAttributesProvider`. Honestly I don't remember in which scenario but I suspect it can happen when you have a `Card` with multiple clickable elements inside. In that case the `Card` is the direct parent of the clickable element and not the `RecyclerView`

Reply:
But in your algorithm here, the child is always the direct child of the parent because either: 

```
        var child: View? = view
        var parent = view.parent // = child.parent
```

or 

```
            child = parent as? View
            parent = parent.parent // = child.parent
```

Reply:
Yes exactly, my intention is that if `Card` has 1 clickable element inside and your action target is that element I can at least go up one level, check if the parent of the Card is the `RecyclerView` and in that case offer as an extra information the `position` of the `Card` holding the clickable element in the `RecyclerView` adapter.
Say: user clicks a button inside a `Card` at position 8 in the RecyclerView without this I would not be able to give that information as the `Button` is not the direct child of the `RecyclerView`.

Reply:
What I still don't understand is why you need to call the `isDirectChildOfRecyclerView`. Because if `parent is RecyclerView`, then the child is automatically a direct child of a RecyclerView.